conans/test/integration/metadata/test_metadata_commands.py

```diff
@@ -198,3 +177,28 @@ def test_no_download_cached(self):
         assert "mylogs2!!!!" in mylogs
         mybuildlogs = load(os.path.join(c2_pkg_metadata_path, "logs", "mybuildlogs.txt"))
         assert "mybuildlogs2!!!!" in mybuildlogs
+
+    def test_upload_ignored_metadata(self, create_conan_pkg):
+        """
+        Upload command should ignore metadata files when passing --metadata=""
+        """
+        client, pid = create_conan_pkg
+
+        self.save_metadata_file(client, "pkg/0.1")
+        self.save_metadata_file(client, f"pkg/0.1:{pid}")
+
+        client.run('upload * --confirm --remote=default --metadata=""')
+        assert "Recipe metadata" not in client.out
+        assert "Package metadata" not in client.out
+
+    def test_upload_ignored_metadata_with_pattern(self, create_conan_pkg):
+        """
+        Upload command should fail when passing --metadata="" and a pattern
+        """
+        client, pid = create_conan_pkg
+
+        self.save_metadata_file(client, "pkg/0.1")
+        self.save_metadata_file(client, f"pkg/0.1:{pid}")
+
+        client.run('upload * --confirm --remote=default --metadata="" --metadata="logs/*"', assert_error=True)
```

Comment:
You should be able to test this without creating packages and without adding metadata to them

Reply:
Done, using only conan export now.